File: app/src/main/java/com/virtualdestination/githubclient/GithubApplication.java
package com.virtualdestination.githubclient;

import android.app.Application;
import android.content.Context;

import com.squareup.picasso.OkHttpDownloader;
import com.squareup.picasso.Picasso;

import butterknife.ButterKnife;

/**
 * Created by Rashid on 26/11/2017.
 */

public class GithubApplication extends Application {


    private static GithubApplication instance;

    public static GithubApplication getInstance() {
        return instance;
    }

    public static Context getContext() {
        return instance.getApplicationContext();
    }

    @Override
    public void onCreate() {
        super.onCreate();

        instance = this;

        ButterKnife.setDebug(BuildConfig.DEBUG);

        Picasso.Builder builder = new Picasso.Builder(this);
        builder.downloader(new OkHttpDownloader(this, Integer.MAX_VALUE));
        Picasso built = builder.build();
        built.setIndicatorsEnabled(false);
        built.setLoggingEnabled(false);
        Picasso.setSingletonInstance(built);


    }

}

File: app/src/main/java/com/virtualdestination/githubclient/GithubApplication.kt
package com.virtualdestination.githubclient

import android.app.Application
import android.content.Context

/**
 * Created by Rashid on 26/11/2017.
 */
class GithubApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    companion object {
        var instance: GithubApplication? = null
            private set
        val context: Context
            get() = instance!!.applicationContext
    }
}